app/scripts/metamask-controller.js

```diff
@@ -852,8 +855,83 @@ export default class MetamaskController extends EventEmitter {
             saveState: async () => {
               await this.keyringController.persistAllKeyrings();
             },
-            removeAccount: async (address) => {
-              await this.removeAccount(address);
+            addAccount: async (_address, origin, handleUserInput) => {
```

Comment:
Was this logic not going to go in the `rpc-methods` package in the snaps repository?

If it's staying here, I'd always vote to minimise the size of the `MetaMaskController` where possible, so could we create a separate file to hold this logic such as `app/scripts/lib/snaps-keyring.ts`?

Reply:
It would also support much easier unit testing which I'd recommend for this amount of logic.

Reply:
We opted to add callbacks into the [eth-snap-keyring](https://github.com/MetaMask/eth-snap-keyring/pull/99) and keep the approval logic in the extension so that we can reuse those callbacks on mobile as well as get translations.

Reply:
I'll try and extract them out